### PR TITLE
feat(shared): support relative links in normalizeRoutePath

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

File: packages/shared/src/utils/normalizeRoutePath.ts [hunk 1]
```diff
@@ -1,15 +1,11 @@
-/**
- * Normalize the given path to the final route path
- */
-export const normalizeRoutePath = (path: string): string => {
-  // split pathname and query/hash
-  const [pathname, ...queryAndHash] = path.split(/(\?|#)/)
+const FAKE_HOST = 'http://.'
 
+export const inferRoutePath = (path: string): string => {
```
Comment (meteorlxy) [inline packages/shared/src/utils/normalizeRoutePath.ts:3]: As it's exported, we'd better split it into a standalone file

File: packages/shared/tests/normalizeRoutePath.spec.ts [hunk 2]
```diff
@@ -15,55 +15,168 @@ const testCases = [
...
-  ['foo.md?bar=baz', 'foo.html?bar=baz'],
-  ['foo.html?bar=baz', 'foo.html?bar=baz'],
-  ['foo?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['foo.md?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['foo.html?bar=baz#qux', 'foo.html?bar=baz#qux'],
-  ['#bar', '#bar'],
-  ['?bar=baz', '?bar=baz'],
-  ['?bar=baz#qux', '?bar=baz#qux'],
+  // relative non index with current
+  ['foo', '/foo.html', '/'],
```
Comment (meteorlxy) [inline packages/shared/tests/normalizeRoutePath.spec.ts:49]: Better to use `[[input1, input2], output]` format, instead of putting the second param  as the third item of the tuple

**Review by meteorlxy — DISMISSED** [review]

File: packages/shared/tests/normalizeRoutePath.spec.ts [hunk 3]
```diff
@@ -2,109 +2,232 @@ import { describe, expect, it } from 'vitest'
...
 
   // unexpected corner cases
-  ['.md', '.html'],
-  ['foo/.md', 'foo/.html'],
-  ['/.md', '/.html'],
-  ['/foo/.md', '/foo/.html'],
+  [['.md'], '.html'],
+  [['foo/.md'], 'foo/.html'],
+  [['/.md'], '/.html'],
+  [['/foo/.md'], '/foo/.html'],
```
Comment (meteorlxy) [inline packages/shared/tests/normalizeRoutePath.spec.ts:185]: Still need cases with current for the last 5 lines

Comment (Mister-Hope) [reply packages/shared/tests/normalizeRoutePath.spec.ts:185]: I actually don't think this is needed,  These are corner cases which we are not expecting users to actually set them in production

Comment (meteorlxy) [reply packages/shared/tests/normalizeRoutePath.spec.ts:185]: They are unexpected, but still should be covered by our unit tests

Comment (meteorlxy) [reply packages/shared/tests/normalizeRoutePath.spec.ts:185]: For example, if a method is going to throw an error with unexpected input, we usually test it with `expect().toThrow()`. For this case, we don't throw errors, but we still need to be aware of those unexpected input.

**Comment by Mister-Hope** [comment]
test added

**Review by meteorlxy — APPROVED** [review]
